src/main/resources/public/index.html

```diff
@@ -9,8 +9,24 @@
 <script src="/js/vendor/jquery-1.12.4.min.js"></script>
 <script src="/js/app.js"></script>
 
+
 <body>
 
+<!-- Modal display code -->
+<div id="myModal" class="modal">
+
+    <!-- Modal content -->
+    <div class="modal-content">

```

Comment:
It's an overlay, like an alert window but it makes the rest of the page non-interactive till the user picks a difficulty. It cant be accessed mid game and currently it changes an invisible field to easy/hard which is sent with the game-state when a user places a ship. I can change it to directly change the game-state though.